### PR TITLE
feat(lib/erasure): construct a trie of chunks of erasure coded values

### PR DESCRIPTION
**Pull request by axaysagathiya**
## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

#3416 
## Primary Reviewer

@kishansagathiya @edwardmack

### CODE REVIEW

File: lib/erasure/erasure.go [hunk 1]
```diff
@@ -99,3 +103,24 @@ func Reconstruct(nValidators uint, chunks [][]byte) ([]byte, error) {
 
 	return res, nil
 }
+
+func ChunksToTrie(chunks [][]byte) (*trie.Trie, error) {
+	chunkTrie := trie.NewEmptyTrie()
+	for i, chunk := range chunks {
+		encodedI, err := scale.Marshal(uint32(i))
```
Comment (jimjbrettj) [inline lib/erasure/erasure.go:110]: This specifically needs to be a uint32?

File: lib/erasure/erasure_test.go [hunk 2]
```diff
@@ -12,100 +12,113 @@ import (
...
-	}{
-		{
-			name:              "1_validators",
-			validators:        1,
-			dataHex:           "0x04020000000000000000000000000000000000000000000000000000000000000000000000000000000000",
-			expectedChunksHex: []string{},
-			expectedError:     errors.New("expected at least 2 validators"),
+var testCases = []struct {
+	name          string
+	validators    uint
```
Comment (jimjbrettj) [inline lib/erasure/erasure_test.go:17]: nit/should this be numValidators?

**Review by jimjbrettj — APPROVED** [review]
lgtm, just small comments

File: lib/erasure/erasure.go (hunk 1 above)
Comment (axaysagathiya) [reply lib/erasure/erasure.go:110]: for i = 6, 
`scale.Marshal(i)` returns [24] but,
`scale.Marshal(uint32(i))` returns [6, 0, 0, 0] which was the same encoding as the Polkadot code.

so, I think it should be uint32.

File: lib/erasure/erasure_test.go (hunk 2 above)
Comment (axaysagathiya) [reply lib/erasure/erasure_test.go:17]: Thank you for your suggestion! 
I've actually incorporated the use of `nValidators` consistently throughout the functions in this package. As a result, I've gone ahead and updated the field name from `validators` to `nValidators` to maintain uniformity.

**Pull request merged**